packages/tests-unit/tests/core/routing/matcher.test.ts

```diff
@@ -266,6 +266,26 @@ describe("handleRedirects", () => {
 
     expect(result).toBeUndefined();
   });
+
+  it("should redirect with + character and query string", () => {
+    const event = createEvent({
+      url: "/foo",
+    });
+
+    const result = handleRedirects(event, [
+      {
```

Comment:
You're not using the correct things here. You should be using the format used in the output from `.next/routes-manifest.json`

Reply:
Nice catch I missed this. Does the unit test run before or after approval? I must have missed the failing test.

Reply:
I've rewrote the test to pass.
`expect(result).toBeUndefined();`

This does feel a little untuitive... The "should redirect matching path" also expects result to be undefined though, so maybe I'm missing something. Is it because it's an internal redirect?

Reply:
It should not be undefined, the problem is what args you're passing to the `handleRedirects` fn.
And the `"should redirect matching path"` has a mistake that i fixed in a recent PR, it should also not return undefined.

The params should contain a regex, and the easiest way to get a correct one is to define your redirect in next config and run build, then get the correct form from `.next/routes-manifest.json`

Reply:
> Does the unit test run before or after approval?
It runs after, that's probably why you missed it

Reply:
Done, thanks!